### PR TITLE
[Task] Update pod route

### PR DESCRIPTION
**Pull request by sufuf3**

### CODE REVIEW

**Comment by John-Lin** [comment]
Conflict

File: src/entity/pod.go [hunk 1]
```diff
@@ -25,21 +25,27 @@ type Container struct {
...
 	// can not validate nil
-	VlanTag   *int32     `bson:"vlanTag" json:"vlanTag" validate:"-"`
-	IPAddress string     `bson:"ipAddress" json:"ipAddress" validate:"required,ipv4"`
-	Netmask   string     `bson:"netmask" json:"netmask" validate:"required,ipv4"`
-	Routes    []PodRoute `bson:"routes,omitempty" json:"routes" validate:"required,dive,required"`
+	VlanTag    *int32         `bson:"vlanTag" json:"vlanTag" validate:"-"`
+	IPAddress  string         `bson:"ipAddress" json:"ipAddress" validate:"required,ipv4"`
+	Netmask    string         `bson:"netmask" json:"netmask" validate:"required,ipv4"`
+	RoutesGw   []PodRouteGw   `bson:"routesGw,omitempty" json:"routesGw" validate:"dive"`
+	RoutesIntf []PodRouteIntf `bson:"routesIntf,omitempty" json:"routesIntf" validate:"dive"`
```
Comment (John-Lin) [inline src/entity/pod.go]: `required,dive,required`

File: src/entity/pod.go [hunk 2]
```diff
@@ -25,21 +25,27 @@ type Container struct {
...
 	IfName string `bson:"ifName" json:"ifName" validate:"required"`
 	// can not validate nil
-	VlanTag   *int32     `bson:"vlanTag" json:"vlanTag" validate:"-"`
-	IPAddress string     `bson:"ipAddress" json:"ipAddress" validate:"required,ipv4"`
-	Netmask   string     `bson:"netmask" json:"netmask" validate:"required,ipv4"`
-	Routes    []PodRoute `bson:"routes,omitempty" json:"routes" validate:"required,dive,required"`
+	VlanTag    *int32         `bson:"vlanTag" json:"vlanTag" validate:"-"`
+	IPAddress  string         `bson:"ipAddress" json:"ipAddress" validate:"required,ipv4"`
+	Netmask    string         `bson:"netmask" json:"netmask" validate:"required,ipv4"`
+	RoutesGw   []PodRouteGw   `bson:"routesGw,omitempty" json:"routesGw" validate:"dive"`
```
Comment (John-Lin) [inline src/entity/pod.go]:
```
required,dive,required
```

Comment (sufuf3) [reply src/entity/pod.go]: I met CI fail if I use `required,dive,required` in validate.

Comment (John-Lin) [reply src/entity/pod.go]: integration test failed.

File: API.md [hunk 3]
```diff
@@ -632,6 +620,11 @@ For each Pod, we have fileds need to handle.
     - vlanTag: the vlan tag for `ifName` interface.
     - ipADdress: the IPv4 address of the `ifName` interface.
     - netmask: the IPv4 netmask of the `ifName` interface.
+    - routesGw: a array of route with gateway (Optional)
+        - dstCIDR: destination network cidr for add IP routing table
```
Comment (John-Lin) [inline API.md]:
```
dstCIDR(required):
```

File: API.md [hunk 4]
```diff
@@ -632,6 +620,11 @@ For each Pod, we have fileds need to handle.
...
+        - gateway: the gateway of the interface subnet
+    - routeIntf: a array of route without gateway (Optional)
+        - dstCIDR: destination network cidr for add IP routing table
```
Comment (John-Lin) [inline API.md]:
```
dstCIDR(required):
```

File: API.md [hunk 5]
```diff
@@ -632,6 +620,11 @@ For each Pod, we have fileds need to handle.
     - vlanTag: the vlan tag for `ifName` interface.
     - ipADdress: the IPv4 address of the `ifName` interface.
     - netmask: the IPv4 netmask of the `ifName` interface.
+    - routesGw: a array of route with gateway (Optional)
+        - dstCIDR: destination network cidr for add IP routing table
+        - gateway: the gateway of the interface subnet
```
Comment (John-Lin) [inline API.md]:
```
gateway(required):
```

**Pull request merged**